lib/index.js

```diff
@@ -213,36 +218,61 @@ Mongoose.prototype.setDriver = function setDriver(driver) {
  * - 'toJSON': `{ transform: true, flattenDecimals: true }` by default. Overwrites default objects to [`toJSON()`](/docs/api.html#document_Document-toJSON), for determining how Mongoose documents get serialized by `JSON.stringify()`
  * - 'toObject': `{ transform: true, flattenDecimals: true }` by default. Overwrites default objects to [`toObject()`](/docs/api.html#document_Document-toObject)
  *
- * @param {String} key
- * @param {String|Function|Boolean} value
+ * @param {String|Object} key The name of the option or a object of multiple key-value pairs
+ * @param {String|Function|Boolean} value The value of the option, unused if "key" is a object
+ * @returns {Mongoose} The used Mongoose instnace
  * @api public
  */
 
 Mongoose.prototype.set = function(key, value) {
   const _mongoose = this instanceof Mongoose ? this : mongoose;
 
-  if (VALID_OPTIONS.indexOf(key) === -1) {
-    throw new Error(`\`${key}\` is an invalid option.`);
+  if (arguments.length === 1 && typeof key !== 'object') {
+    return _mongoose.options[key];
   }
 
-  if (arguments.length === 1) {
-    return _mongoose.options[key];
+  let options = {};
+
+  if (arguments.length === 2) {
+    options = { [key]: value };
   }
 
-  _mongoose.options[key] = value;
+  if (arguments.length === 1 && typeof key === 'object') {
+    options = key;
+  }
 
-  if (key === 'objectIdGetter') {
-    if (value) {
-      Object.defineProperty(mongoose.Types.ObjectId.prototype, '_id', {
-        enumerable: false,
-        configurable: true,
-        get: function() {
-          return this;
-        }
-      });
-    } else {
-      delete mongoose.Types.ObjectId.prototype._id;
+  // array for errors to collect all errors for all key-value pairs, like ".validate"
+  const errors = [];
+
+  for (const [k, v] of Object.entries(options)) {
```

Comment:
nitpick: `k`, `v` aren't very descriptive, I'd go for something like `key`, `value` which would be more readable.
```suggestion
  for (const [key, value] of Object.entries(options)) {
```

Reply:
> nitpick: k, v aren't very descriptive, I'd go for something like key, value which would be more readable.

the problem with the suggested are that those names are already taken, without renaming the function parameters

Reply:
I agree that we shouldn't shadow the original key/value.

We can use special identifiers for them and avoid conflict. What do you think of `optionKey` and `optionValue`?

Reply:
sure

Reply:
done in 28b12e26670571088f1b634fbd19adbfebb8f7db